src/controllers/update/overseas.entity.query.controller.ts

```diff
@@ -32,7 +33,9 @@ export const post = async (req: Request, res: Response, next: NextFunction) => {
 
     const entityNumber = req.body[EntityNumberKey];
     const companyProfile = await getCompanyProfile(req, entityNumber);
-    if (!companyProfile) {
+    const privateOeDetails = await getPrivateOeDetails(req, entityNumber);
```

Comment:
Only fetch if companyProfile is not already fetched.

Reply:
Make sure you fetch the company profile first and handle any errors. Only then attempt the email fetch.

Reply:
Now calling `getPrivateOeDetails` only if `companyProfile` exists.